### PR TITLE
Implement lockfile tests

### PR DESCRIPTION
**Pull request by maciektr**
commit-id:80d6f606

---

**Stack**:
- #781
- #800
- #780 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: scarb/src/resolver/mod.rs [hunk 1]
```diff
@@ -247,8 +257,16 @@ mod tests {
     }
 
     fn resolve(
+        registry: MockRegistry,
+        roots: Vec<(&[ManifestDependency], PackageId)>,
+    ) -> anyhow::Result<Resolve> {
+        resolve_with_lock(registry, roots, &[])
+    }
```
Comment (mkaput) [inline scarb/src/resolver/mod.rs:264]: dead code?

File: scarb/src/resolver/mod.rs [hunk 2]
```diff
@@ -247,8 +257,16 @@ mod tests {
     }
 
     fn resolve(
+        registry: MockRegistry,
+        roots: Vec<(&[ManifestDependency], PackageId)>,
+    ) -> anyhow::Result<Resolve> {
```
Comment (mkaput) [inline scarb/src/resolver/mod.rs:262]:
```suggestion
    ) -> Result<Resolve> {
```

File: scarb/src/resolver/mod.rs [hunk 3]
```diff
@@ -247,8 +257,16 @@ mod tests {
...
+        resolve_with_lock(registry, roots, &[])
+    }
+
+    fn resolve_with_lock(
         mut registry: MockRegistry,
         roots: Vec<(&[ManifestDependency], PackageId)>,
+        locks: &[PackageLock],
     ) -> anyhow::Result<Resolve> {
```
Comment (mkaput) [inline scarb/src/resolver/mod.rs:270]: `anyhow::` -> ``

File: scarb/src/resolver/mod.rs [hunk 4]
```diff
@@ -265,7 +283,10 @@ mod tests {
             })
             .collect_vec();
 
-        let lockfile = Lockfile::new(vec![].into_iter());
+        let lockfile = Lockfile {
+            version: Default::default(),
+            packages: BTreeSet::from_iter(locks.iter().cloned()),
```
Comment (mkaput) [inline scarb/src/resolver/mod.rs:288]:
```suggestion
            packages: locks.iter().cloned().collect(),
```

btw why can't you use `Lockfile::new` here anymore?

**Review by mkaput — APPROVED** [review]

File: scarb/src/resolver/mod.rs (hunk 1 above)
Comment (maciektr) [reply scarb/src/resolver/mod.rs:264]: It's actually used exactly once, here: https://github.com/software-mansion/scarb/blob/7a5356a51fd4090f212f62a49b2c0b44c22d37dc/scarb/src/resolver/mod.rs#L616C2-L616C2

**Review by mkaput — APPROVED** [review]

**Pull request merged**